src/WebJobs.Extensions.DurableTask/DurableHttpRequest.cs

```diff
@@ -169,7 +169,8 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                 }
                 else
                 {
-                    // Don't know how to serialize this - use default behavior
+                    // Don't know how to serialize this - use default behavior, forcing TypeNameHandling.Objects to correctly serialize ITokenSource
+                    serializer.TypeNameHandling = TypeNameHandling.Objects;
```

Comment:
This serializer might be reused later (not sure about how it works internally with Newtonsoft. Might be safer to store the current value of `serializer.TypeNameHandling` before we assign it to `TypeNameHandling.Objects`, sot that after calling `serializer.Serialize(writer, value)` we can reassign it to it's old value.

Reply:
Isn't this a non-issue because the serializer is passed in as a parameter?

Reply:
Newtonsoft manages that serializer passed in as a parameter. I am not positive, but the serializer they give us MAY be used in more than just this one place.